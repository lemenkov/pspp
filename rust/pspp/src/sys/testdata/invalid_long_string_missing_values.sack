# File header.
"$FL2"; s60 "$(#) SPSS DATA FILE PSPP synthetic test file";
2; # Layout code
8; # Nominal case size
0; # Not compressed
0; # Not weighted
0; # 0 cases.
100.0; # Bias.
"01 Jan 11"; "20:53:52";
"PSPP synthetic test file: "; i8 244; i8 245; i8 246; i8 248; s34 "";
i8 0 *3;

# One numeric variable.
2; 0; 0; 0; 0x050800 *2; s8 "NUM1";

# Long string variables that will have missing values added with a
# later record.
2; 9; 0; 0; 0x010900 *2; s8 "STR1";
2; -1; 0; 0; 0; 0; s8 "";
2; 10; 0; 0; 0x010a00 *2; s8 "STR2";
2; -1; 0; 0; 0; 0; s8 "";
2; 11; 0; 0; 0x010b00 *2; s8 "STR3";
2; -1; 0; 0; 0; 0; s8 "";
2; 7; 0; 0; 0x010700 *2; s8 "STR4";
2; -1; 0; 0; 0; 0; s8 "";

# Machine integer info record.
7; 3; 4; 8; 1; 2; 3; -1; 1; 1; ENDIAN; 1252;

# Machine floating-point info record.
7; 4; 8; 3; SYSMIS; HIGHEST; LOWEST;

# Long string variable missing values record.
7; 22; 1; COUNT (
# Zero missing values (not allowed) for STR1 .
COUNT("STR1"); i8 >>0<<; 8;

# Four missing values (not allowed) for STR2.
COUNT("STR2"); i8 4; 8;
"abcdefgh"; "ijklmnop"; "qrstuvwx"; "yz012345";

# Missing values for unknown variable
COUNT(>>"Nonexistent"<<); i8 1; 8; "abcdefgh";

# Missing values for numeric variable
COUNT(>>"NUM1"<<); i8 1; 8; "abcdefgh";

# Too long missing value
COUNT("STR3"); i8 1; >>COUNT("abcdefghijkl")<<;

# Buggy way that this was written in old PSPP, with a length
# before each missing value instead of just once.
COUNT("STR3"); i8 2; 8; "ABCDEFGH"; >>8<<; "IJKLMNOP";

# Invalid 8-byte missing value for 7-byte string.
COUNT("STR4"); i8 1; 8; "ABCDEFGH";
);

# Character encoding record.
7; 20; 1; 12; "windows-1252";

# Dictionary termination record.
999; 0;
