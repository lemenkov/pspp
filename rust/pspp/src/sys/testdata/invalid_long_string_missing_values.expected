Warning at file offsets 0x238 to 0x2f2: In long string missing values record: Value length at offset 0x2a8 is 12 instead of the expected 8.

File header claims 8 variable positions but 9 were read from file.

Long string missing values record says variable STR2 has 4 missing values, but only 1 to 3 missing values are allowed.

Unknown variable name "Nonexistent" in long string missing value record.

Invalid long string missing value for numeric variable NUM1.

Invalid long string missing value for 7-byte string variable STR4.

╭──────────────────────┬────────────────────────╮
│       Created        │    01-JAN-2011 20:53:52│
├──────────────────────┼────────────────────────┤
│Writer Product        │PSPP synthetic test file│
│       Version        │1.2.3                   │
├──────────────────────┼────────────────────────┤
│       Compression    │None                    │
│       Number of Cases│                       0│
╰──────────────────────┴────────────────────────╯

╭─────────┬──────────────────────────────╮
│Label    │PSPP synthetic test file: ôõöø│
│Variables│                             5│
╰─────────┴──────────────────────────────╯

                                                         Variables
╭────┬────────┬─────┬─────────────────┬─────┬─────┬─────────┬────────────┬────────────┬──────────────────────────────────╮
│    │Position│Label│Measurement Level│ Role│Width│Alignment│Print Format│Write Format│          Missing Values          │
├────┼────────┼─────┼─────────────────┼─────┼─────┼─────────┼────────────┼────────────┼──────────────────────────────────┤
│num1│       1│     │                 │Input│    8│Right    │F8.0        │F8.0        │                                  │
│str1│       2│     │Nominal          │Input│    9│Left     │A9          │A9          │                                  │
│str2│       3│     │Nominal          │Input│   10│Left     │A10         │A10         │"abcdefgh"; "ijklmnop"; "qrstuvwx"│
│str3│       4│     │Nominal          │Input│   11│Left     │A11         │A11         │"ABCDEFGH"; "IJKLMNOP"            │
│str4│       5│     │Nominal          │Input│    7│Left     │A7          │A7          │                                  │
╰────┴────────┴─────┴─────────────────┴─────┴─────┴─────────┴────────────┴────────────┴──────────────────────────────────╯
